### PR TITLE
Index of predictions

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#256 
#### What does this implement/fix? Explain your changes.
+ `position` method
+ `positions` attribute
+ `label_map` attribute

This pull request adds `position` method which may help with finding the indexes of observations in `predict_vector` which made TP, TN, FP, FN according to the `actual_vector`.
After first call, this method will save `positions` dictionary in memory in order to avoid recalculation.
`positions` is a dictionary which contains list of indexes for each situation(TP,TN,FP,FN) divided in classes, it should be like:
```
{class_1:{'TP':[1, 0], 'FP':[2], 'TN':[3, 4], 'FN':[5]}, class_2: ..., ...}
```
There was some problem with this new feature and `relabel` method so `label_map` is a new dictionary for class that tracks the relabels from the beginning of the class and matches new labels to first class names.
Now after each call for `relabel`, `positions` will be `None` and after relabeling the classes, new generated `positions` after `position` call may have proper names respected to new labels.

### CODE REVIEW

File: Test/error_test.py [hunk 1]
```diff
@@ -20,6 +20,11 @@
 Traceback (most recent call last):
         ...
 pycm.pycm_obj.pycmVectorError: Input vectors must have same length
+>>> cm_6 = ConfusionMatrix(matrix={0:{0:2,1:50,2:6},1:{0:5,1:50,2:3},2:{0:1,1:7,2:50}})
+>>> cm_6.position()
+Traceback (most recent call last):
+        ...
+pycm.pycm_error.pycmVectorError: Input vectors are empty
```
Comment (sadrasabouri) [inline Test/error_test.py:27]: @sepandhaghighi 
If it is necessary, I can add new error message instead `Input vectors are empty`.

**Comment by sepandhaghighi** [comment]
@sadrasabouri I think you should consider relabeling condition ;-)

File: Test/function_test.py [hunk 2]
```diff
@@ -342,6 +342,55 @@
...
+>>> POS[2]['TN']
+[1, 4, 6, 9]
+>>> POS[2]['FP']
+[5, 10]
+>>> POS[2]['FN']
+[0, 3, 7]
+>>> y_actu = [0, 0, 1, 1, 0]
+>>> y_pred = [0, 1, 1, 0, 0]
+>>> cm2 = ConfusionMatrix(actual_vector=y_actu, predict_vector=y_pred)
+>>> POS = cm.position()
```
Comment (sepandhaghighi) [inline Test/function_test.py:376]: @sadrasabouri I think here you should use `cm2`

File: Test/function_test.py [hunk 3]
```diff
@@ -342,6 +342,55 @@
...
+[]
+>>> POS[1]['TP']
+[6]
+>>> POS[1]['TN']
+[0, 1, 2, 4, 7, 8, 9, 11]
+>>> POS[1]['FP']
+[3]
+>>> POS[1]['FN']
+[5, 10]
+>>> cm2.relabel({0:'L1',1:'L2'})
```
Comment (sepandhaghighi) [inline Test/function_test.py:393]: Please read relabeled classes after this line, for example :
```pycon
>>> POS = cm.position()
>>> POS["L1"]["TP"]
```
You should update `self.positions` in `relabel` method

Comment (sadrasabouri) [reply Test/function_test.py:409]: Ok
I'll handle it.

File: Test/error_test.py (hunk 1 above)
Comment (sepandhaghighi) [inline Test/error_test.py:27]: @sadrasabouri Thanks for your effort ;-)
My last comment for this PR : 
Please change this message to something like -- > `This option only works in vector mode`

Best Regards

**Comment by sepandhaghighi** [comment]
@sadrasabouri Thanks for your effort ;-) LGTM
@alirezazolanvari Please review this PR

**Pull request merged**